myhoard/binary_io_slice.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) 2024 Aiven, Helsinki, Finland. https://aiven.io/
+from types import TracebackType
+from typing import BinaryIO, Type, Iterable, Iterator
+
+
+class BinaryIOSlice(BinaryIO):
+    def __init__(self, max_file_size: int, stream: BinaryIO):
+        super().__init__()
+        self._size_remaining = max_file_size
+        self.stream = stream
+
+    def read(self, __n: int = ...) -> bytes:
```

Comment:
Performance Nerd Comment...

In below, I would avoif the `if` by introducing a new variable.  This new variable would be `m = min(__n, self._size_remaining)`.  I would then have a single call to `read`: `result = self.stream.read(m)`.

Some implementation of `min` avoids the `if`, so this might save some CPU cycles.  And IMHO this is cleaner code.